### ISSUE TITLE
Feature Request: User Management / Authentication

### ISSUE THREAD

**Issue by PeterXQChen** [Maker-Management-Platform/docs#13]
User management or some sort of an authentication system would be great since I plan on exposing this to the public.

**Comment by dockrbyter** [comment]
That's sad. I would love to run this on my server, but it would need to be compatible with authentik... 🫤

**Comment by EduardoOliveira** [comment]
@dockrbyter there is a user on discord that implemented Auth at the http proxy level. That might be a possible solution

**Comment by dockrbyter** [comment]
No, not realy. Like I've said, I'm looking for something that supports SSO with https://goauthentik.io
The reason for me to use your - or any similiar software would be to share my STLs with my friends which already have an account on my server. So yeah, I see great potential in your software, but sadly useless for me at this point of development.
Good luck with your project

**Comment by EduardoOliveira** [comment]
@dockrbyter on possible solution, with MMP as is:
- spin up an MMP instance
- spin up a caddy container configured as a reverse proxy enforcing a basic auth authentication, https://caddyserver.com/docs/caddyfile/directives/basicauth
- configure goauthentik following the *arr example https://docs.goauthentik.io/integrations/services/sonarr/

Really hope this research helps you find MMP less useless

Given my limited availability, I think there are higher priority features then implementing an auth system within MMP, but I would be happy to accept code contributions implementing them.
